custom_model_runner/datarobot_drum/drum/data_marshalling.py

```diff
@@ -0,0 +1,164 @@
+from typing import Any, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from datarobot_drum.drum.common import TargetType
+from datarobot_drum.drum.enum import LOGGER_NAME_PREFIX, REGRESSION_PRED_COLUMN
+from datarobot_drum.drum.exceptions import DrumCommonException
+
+logger = logging.getLogger(LOGGER_NAME_PREFIX + "." + __name__)
+
+
+def marshal_predictions(
+    request_labels: List[str],
+    predictions: np.array,
+    target_type: TargetType,
+    model_labels: Optional[List[Any]] = None,
+) -> pd.DataFrame:
+    """
+    :param request_labels: These labels are sent in from DataRobot in every prediction request
+    :param predictions: The predictions returned from the custom code
+    :param target_type: Which sort of target the model is supposed to work on
+    :param model_labels: Optionally, what labels the model thinks the predictions should use
+    :return:
+    Predictions DataFrame
+    """
+    predictions = _validate_dimensionality_and_type(predictions)
+    if target_type.value in TargetType.CLASSIFICATION.value:
+        return _classification_marshal_preds(predictions, request_labels, model_labels)
+    elif target_type.value in TargetType.SINGLE_COL.value:
+        return _single_col_marshal_preds(predictions)
+    return predictions
+
+
+def _marshal_labels(request_labels: List[str], model_labels: List[Any]):
+    if model_labels is not None:
+        if set(_standardize(r_l) for r_l in request_labels) == set(
+            _standardize(m_l) for m_l in model_labels
+        ):
+            return _order_by_float(request_labels, model_labels)
+
+        raise DrumCommonException(
+            "Expected predictions to have columns {}, but encountered {}".format(
+                request_labels, model_labels
+            )
+        )
+    return request_labels
+
+
+def _order_by_float(expected_labels, actual_labels):
+    """
+    Match the order of actual labels to the values in expected labels
+    Given both can be cast to floats
+    >>> _order_by_float(["1.0", "2.4", "0.4", "1.4"],  [2.4, 1.0, 0.4, 1.4])
+    ['2.4', '1.0', '0.4', '1.4']
+    """
+
+    def get_corresponding_expected_label(a_l):
+        for e_l in expected_labels:
+            if _standardize(a_l) == _standardize(e_l):
+                return e_l
+
+    return [get_corresponding_expected_label(_l) for _l in actual_labels]
+
+
+def _standardize(label):
+    # First, hope it's a float
+    try:
+        return float(label)
+    except ValueError:
+        pass
+
+    # Maybe if its a boolean we can make it floaty anyways
+    try:
+        return float(strtobool(label))
+    except ValueError:
+        pass
+
+    # Okay lets just do a str.lower
+    assert isinstance(label, str)
+    return label.lower()
+
+
+def _classification_marshal_preds(predictions, request_labels, model_labels):
+    request_labels = _marshal_labels(request_labels, model_labels)
+    predictions = _duplicate_1d_probabilities(predictions, request_labels)
```

Comment:
nit: _infer_negative_class_probabilities or something like that is probably a better name, but not enough to warrant rerunning tests